No lint issues, but gradual fails on empty compare

See workflow run here: https://github.com/pboling/active_security/actions/runs/9799138608/job/27058821636

```
Running RuboCop Gradual...
..................................................
Found 0 files with 0 issue(s).
Processing results...
RuboCop Gradual is complete!
RuboCop Gradual failed!

Unexpected Changes!

RuboCop Gradual lock file is outdated, to fix this message:
- Run `rubocop-gradual` locally and commit the results

`.rubocop_gradual.lock` diff:

-{
-
-}

Error: Process completed with exit code 1.
```

There is no `rubocop_gradual.lock` in the repo because there are no lint issues at all.  When I run locally I get:
```
❯ bundle exec rake rubocop_gradual
Running RuboCop Gradual...
..................................................
Found 0 files with 0 issue(s).
Processing results...
RuboCop Gradual is complete!
```
When I run the same `check` command locally I have the same problem.
```
bundle exec rake rubocop_gradual:check
Running RuboCop Gradual...
...............................................
Found 0 files with 0 issue(s).
Processing results...
RuboCop Gradual is complete!

Unexpected Changes!

RuboCop Gradual lock file is outdated, to fix this message:
- Run `rubocop-gradual` locally and commit the results

`.rubocop_gradual.lock` diff:

-{
-
-}
```

This used to work, so it must be due to a recent change somewhere.

--- a/lib/rubocop/gradual/process.rb
+++ b/lib/rubocop/gradual/process.rb
@@ -35,7 +35,10 @@ module RuboCop
       private
 
       def fail_with_outdated_lock?(diff)
-        Configuration.mode == :check && diff.state != :no_changes
+        return false if Configuration.mode != :check
+        return false if diff.state == :complete && old_results.nil?
+
+        diff.state != :no_changes
       end
 
       def sync_lock_file(diff)

--- a/spec/rubocop/gradual_spec.rb
+++ b/spec/rubocop/gradual_spec.rb
@@ -167,6 +167,25 @@ RSpec.describe RuboCop::Gradual, :aggregate_failures do
     include_examples "error with --check option", with_extended_steps: false
   end
 
+  context "when no lock file and no issues found" do
+    let(:options) { %w[--config security_only_rubocop.yml --gradual-file] }
+    let(:actual_lock_path) { File.expand_path("no.lock") }
+
+    it "removes file" do
+      expect(gradual_cli).to eq(0)
+      expect($stdout.string).to include("RuboCop Gradual is complete!")
+    end
+
+    context "with --check option" do
+      let(:options) { super().unshift("--check") }
+
+      it "returns success and doesn't update file" do
+        expect(gradual_cli).to eq(0)
+        expect($stdout.string).to include("RuboCop Gradual is complete!")
+      end
+    end
+  end
+
   context "with --autocorrect option" do
     let(:options) { %w[--autocorrect --gradual-file] }
     let(:actual_lock_path) { File.expand_path("full.lock") }

